shacl/offendingTriples.py

```diff
@@ -0,0 +1,177 @@
+# This module is used by _pyshacl.py, a thin wrapper of pyshacl
+# to find offending triples for each reported constraint violation.
+
+import argparse
+from rdflib import Graph, Namespace, URIRef, Literal
+from rdflib.plugins.sparql import prepareQuery
+import sys
+
+sys.path.append('..')
+from bricksrc.namespaces import RDF, RDFS, BRICK, BSH, SH, SKOS
+
+global result_graph
```

Comment:
Can you refactor this to not use global variables? I think this could do nicely as a class that wraps up the logic

Reply:
Done